tests/bundle/EventListener/ConditionallyRemoveVaryHeaderListenerTest.php

```diff
@@ -0,0 +1,97 @@
+{
+    /** @var EzSystems\PlatformHttpCacheBundle\EventListener\ConditionallyRemoveVaryHeaderListener */
+    public $conditionallyRemoveVaryHeaderListener;
+
+    protected function setUp(): void
+    {
+        $this->conditionallyRemoveVaryHeaderListener = new ConditionallyRemoveVaryHeaderListener(['testroute1', 'testroute2']);
+    }
+
+    /**
+     * @return iterable<array{varyHeaders: string[], expectedVaryHeaders: string[]}>
+     */
+    public function onKernelResponseProvider(): iterable
+    {
+        return [
+            [
+                'varyHeaders' => ['Cookie', 'vtest'],
+                'expectedVaryHeaders' => ['vtest'],
+            ],
+            [
+                'varyHeaders' => ['Authorization'],
+                'expectedVaryHeaders' => [],
+            ],
+            [
+                'varyHeaders' => ['cookie'],
+                'expectedVaryHeaders' => [],
+            ],
+            [
+                'varyHeaders' => ['authorization'],
+                'expectedVaryHeaders' => [],
+            ],
+            [
+                'varyHeaders' => ['Cookie', 'vtest'],
+                'expectedVaryHeaders' => ['vtest'],
+            ],
+            [
+                'varyHeaders' => ['vtest', 'Cookie'],
+                'expectedVaryHeaders' => ['vtest'],
+            ],
+            [
+                'varyHeaders' => ['cookie', 'vtest'],
+                'expectedVaryHeaders' => ['vtest'],
+            ],
+            [
+                'varyHeaders' => ['cookie', 'vtest', 'vtest2', 'vtest3', 'authorization'],
+                'expectedVaryHeaders' => ['vtest', 'vtest2', 'vtest3'],
+            ],
+            [
+                'varyHeaders' => ['cookie', 'vtest', 'vtest2', 'vtest3', 'authorization', 'vtest4'],
+                'expectedVaryHeaders' => ['vtest', 'vtest2', 'vtest3', 'vtest4'],
+            ],
+        ];
+    }
+
+    /**
+     * @dataProvider onKernelResponseProvider
+     *
+     * @param string[] $varyHeaders
+     * @param string[] $expectedVaryHeaders
+     */
+    public function testOnKernelResponse($varyHeaders, $expectedVaryHeaders)
+    {
+        $request = $this->createMock(Request::class);
+        $request->method('get')
+            ->willReturn('testroute1');
+
+        $response = new Response('test content', 200, ['vary' => $varyHeaders]);
+
+        $kernel = $this->createMock(HttpKernelInterface::class);
+        $event = new ResponseEvent($kernel, $request, HttpKernelInterface::MAIN_REQUEST, $response);
+
+        $this->conditionallyRemoveVaryHeaderListener->onKernelResponse($event);
+        $this->assertNotContains('cookie', $response->headers->all('vary'));
+        $this->assertNotContains('Cookie', $response->headers->all('vary'));
+        $this->assertNotContains('authorization', $response->headers->all('vary'));
+        $this->assertNotContains('Authorization', $response->headers->all('vary'));
```

Comment:
CS: static method should be referred to in a static way
```suggestion
        self::assertNotContains('cookie', $response->headers->all('vary'));
        self::assertNotContains('Cookie', $response->headers->all('vary'));
        self::assertNotContains('authorization', $response->headers->all('vary'));
        self::assertNotContains('Authorization', $response->headers->all('vary'));
```